### PR TITLE
First poc

### PR DESCRIPTION
**Pull request by almarklein**
Screenshot of the `slicer_with_3_views.py` exampe:
afbeelding

### CODE REVIEW

File: dash_3d_viewer/slicer.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        )
+        # Create the stores that we need (these must be present in the layout)
+        self.stores = [
+            Store(id=self._subid("slice-index"), data=volume.shape[self._axis] // 2),
+            Store(id=self._subid("_requested-slice-index"), data=0),
+            Store(id=self._subid("_slice-data"), data=""),
+        ]
+
+        self._create_server_callbacks(app)
+        self._create_client_callbacks(app)
```
Comment (almarklein) [inline dash_3d_viewer/slicer.py:94]: Right now I create the callbacks when this object gets initialized. According to the Dash docs "Callbacks require their Inputs, States, and Output to be present in the layout" unless `app.config.suppress_callback_exceptions = True`. I was wondering why this does work then?

**Comment by emmanuelle** [comment]
Really cool! I'll read at least some of the code before our meeting this afternoon. Could you please add a minimal `setup.py` so that we can install the package locally to run the examples?

**Comment by emmanuelle** [comment]
@nicolaskruchten :eyes:

**Comment by almarklein** [comment]
> Could you please add a minimal setup.py 

Will do. Though if you run `python examples/xxxx.py` from the root, it should also work, I think.

**Comment by almarklein** [comment]
> Though if you run python examples/xxxx.py from the root, it should also work, I think.

Nope, the current dir is only in PYTHONPATH when running interactively. Anyway, setup.py is added, so you can do e.g. `pip install -e .`

File: dash_3d_viewer/utils.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import PIL.Image
+import skimage
+from plotly.utils import ImageUriValidator
+
+
+def gen_random_id(n=6):
+    return "".join(random.choice("abcdefghijklmnopqrtsuvwxyz") for i in range(n))
+
+
+def img_array_to_uri(img_array):
```
Comment (emmanuelle) [inline dash_3d_viewer/utils.py:12]: we also have https://github.com/plotly/plotly.py/blob/master/packages/python/plotly/plotly/express/_imshow.py#L27 which has a number of useful options (compression, png or jpg, PIL dependency optional only). The function is not part of the plotly public API but we could (I think we should) make it part of the API.

File: dash_3d_viewer/slicer.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        )
+        fig.update_yaxes(
+            showgrid=False,
+            scaleanchor="x",
+            range=(slice_size[1], 0),  # todo: allow flipping x or y
+            showticklabels=False,
+            zeroline=False,
+        )
+        # Add an empty layout image that we can populate from JS.
+        fig.add_layout_image(
```
Comment (emmanuelle) [inline dash_3d_viewer/slicer.py:57]: I know we sent you code where the image was added as a layout image but the more "modern" way of doing this is to use an Image trace like in https://github.com/plotly/dash-sample-apps/blob/master/apps/dash-covid-xray/app.py#L68 and https://github.com/plotly/dash-sample-apps/blob/master/apps/dash-covid-xray/app.py#L418. This way you can get hover and click events on image pixels, which is not the case with a layout image

File: dash_3d_viewer/slicer.py (hunk 1 above)
Comment (almarklein) [reply dash_3d_viewer/slicer.py:94]: Nope, this is fine. I interpreted the docs the wrong way. I made a [pr](https://github.com/plotly/dash-docs/pull/1007) to (hopefully) clarify the text a bit.

File: dash_3d_viewer/utils.py (hunk 2 above)
Comment (almarklein) [reply dash_3d_viewer/utils.py:12]: That would be nice! I'm a bit reluctant to use a private function in a private module of another package :) so for now I added  a note in the code to use it once it become public.

File: dash_3d_viewer/slicer.py (hunk 3 above)
Comment (almarklein) [reply dash_3d_viewer/slicer.py:57]: I saw that the two examples used different methods, and I assumed that layout images where easier, but will change this :) I think I'll do that in a new PR. Are there any other differences? E.g. I saw that with layout images one can create a stack of them and use alpha blending to overlay e.g. segmentation results. Is that possible with image traces as well?

**Pull request merged**